packages/data-models/src/request-models/donation-request-custom-fields.ts

```diff
@@ -4,15 +4,33 @@ export class DonationRequestCustomFields {
   referrer?: string;
   fee_amount_covered?: number;
 
+  /**
+   * The origin is made up of campaign / ABTest information that the user originated from.
+   *
+   * The field is a string so can be expanded on,
+   * but generally has the format `{Source}-{Test Name}-{Variant Name}`, ie
```

Comment:
There's no constraints on the `origin` field, it's just a freeform string. This is the format that we're using for the donation banner, but it could be anything really, as long as we have some identifier to query for in Civi.

Reply:
I think I'm going to update this JSDoc to be less specific about the format since it can be anything.